### PR TITLE
[BugFix] fix empty key error in to_json

### LINKED ISSUE #27527: Cast map to json failed when empty key exists
### Steps to reproduce the behavior (Required)

- Current: `MAP` allows empty key string, but `JSON` not allow. 
- Fix: `to_json` function should skip the empty key

```sql
select to_json(map{'': 1, 'a': 2});
```

### StarRocks version (Required)
- 3.0, 3.1

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #27527

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/exprs/cast_expr_json.cpp [hunk 1]
```diff
@@ -40,7 +42,14 @@ class CastColumnItemVisitor final : public ColumnVisitorAdapter<CastColumnItemVi
...
     static Status cast_datum_to_json(const ColumnPtr& col, int row, const std::string& name, vpack::Builder* builder) {
         CastColumnItemVisitor visitor(row, name, builder);
-        return col->accept(&visitor);
+        try {
+            return col->accept(&visitor);
+        } catch (const arangodb::velocypack::Exception& e) {
+            std::string data = col->debug_item(row);
+            LOG(WARNING) << "cast to json failed: error_code=" << e.errorCode() << ", error_msg=" << e.what()
+                         << ", data=" << data;
+            return Status::DataQualityError("cast to json failed: " + data);
```
Comment (jaogoy) [inline be/src/exprs/cast_expr_json.cpp:51]: why not put the whole error msg in the returning Status

**Review by fzhedu — APPROVED** [review]
LGTM

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

File: be/src/exprs/cast_expr_json.cpp (hunk 1 above)
Comment (mofeiatwork) [reply be/src/exprs/cast_expr_json.cpp:51]: it's too complicated for normal user.

**Review by Seaven — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-3.1

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**